rhp/v2/contracts.go

```diff
@@ -69,52 +69,51 @@ func ContractRenewalCost(fc types.FileContract, contractFee types.Currency) type
 	return fc.ValidRenterPayout().Add(contractFee).Add(contractTax(fc))
 }
 
-// PrepareContractRenewal constructs a contract renewal transaction.
-func PrepareContractRenewal(currentRevision types.FileContractRevision, renterKey types.PrivateKey, hostKey types.PublicKey, renterPayout types.Currency, endHeight uint64, host HostSettings, refundAddr types.Address) types.FileContract {
-	// calculate "base" price and collateral -- the storage cost and collateral
-	// contribution for the amount of data already in contract. If the contract
-	// height did not increase, basePrice and baseCollateral are zero.
+// ContractBaseCosts returns the base price and collateral for a contract.
+func ContractBaseCosts(fc types.FileContract, s HostSettings, endHeight uint64) (types.Currency, types.Currency) {
 	var basePrice, baseCollateral types.Currency
-	if contractEnd := uint64(endHeight + host.WindowSize); contractEnd > currentRevision.WindowEnd {
-		timeExtension := uint64(contractEnd - currentRevision.WindowEnd)
-		basePrice = host.StoragePrice.Mul64(currentRevision.Filesize).Mul64(timeExtension)
-		baseCollateral = host.Collateral.Mul64(currentRevision.Filesize).Mul64(timeExtension)
-	}
-
-	// estimate collateral for new contract
-	var newCollateral types.Currency
-	if costPerByte := host.UploadBandwidthPrice.Add(host.StoragePrice).Add(host.DownloadBandwidthPrice); !costPerByte.IsZero() {
-		bytes := renterPayout.Div(costPerByte)
-		newCollateral = host.Collateral.Mul(bytes)
-	}
 
-	// the collateral can't be greater than MaxCollateral
-	totalCollateral := baseCollateral.Add(newCollateral)
-	if totalCollateral.Cmp(host.MaxCollateral) > 0 {
-		totalCollateral = host.MaxCollateral
```

Comment:
This is very important. I don't think we can get rid of it. Without it, we will potentially fail to refresh and renew contracts that would otherwise succeed.

Reply:
Hm well I know that but I figured if you pass the `hostCollateral` as an argument you don't want to silently cap it. We cap it to `MaxCollateral` in the autopilot. I re-added it though.